barnetrygd/src/main/kotlin/no/nav/familie/kontrakter/ba/søknad/VersjonertBarnetrygdSøknad.kt

```diff
@@ -0,0 +1,53 @@
+package no.nav.familie.kontrakter.ba.søknad
+
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonDeserializer
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import no.nav.familie.kontrakter.felles.søknad.BaksSøknadBase
+import no.nav.familie.kontrakter.felles.søknad.MissingVersionImplementationException
+import no.nav.familie.kontrakter.felles.søknad.UnsupportedVersionException
+import no.nav.familie.kontrakter.ba.søknad.v7.Søknad as BarnetrygdSøknadV7
+import no.nav.familie.kontrakter.ba.søknad.v8.Søknad as BarnetrygdSøknadV8
+import no.nav.familie.kontrakter.ba.søknad.v9.BarnetrygdSøknad as BarnetrygdSøknadV9
+
+class VersjonertBarnetrygdSøknadDeserializer : JsonDeserializer<VersjonertBarnetrygdSøknad>() {
+    /**
+     * @throws UnsupportedVersionException dersom `kontraktVersjon` ikke er støttet.
+     * @throws MissingVersionImplementationException `kontraktVersjon` ikke finnes i JSON-string.
+     */
+    override fun deserialize(
+        p: JsonParser,
+        ctxt: DeserializationContext,
+    ): VersjonertBarnetrygdSøknad {
+        val node: JsonNode = p.codec.readTree(p)
+        val versjon =
+            node.get("kontraktVersjon")?.asInt()
+                ?: throw UnsupportedVersionException("JSON-string mangler feltet 'kontraktVersjon' og kan ikke deserialiseres. $node")
```

Comment:
Burde den heller bli satt til null, slik at man heller kaster MissingVersionImplementationException i steget under?

Reply:
Grunnen til at det er gjort slik ☝️ Er for å håndtere `UnsupportedVersionException` og `MissingVersionImplementationException` forskjellig i familie-integrasjoner. Kan se at navnene kanskje burde justeres for å bedre reflektere hvilken feil de representerer 🤔

Reply:
Slik jeg leser koden, så vil MissingVersionImplementationException aldri forekomme pga UnsupportedVersionException

Reply:
men jeg forstår nå. Den vil kaste når nr 10 kommer og man har glemt å lage en

Reply:
Ja, stemmer 👍